### PR TITLE
feat(slider): added new slider component

### PR DESCRIPTION
**Pull request by cscheffauer**
# Description

- Added new slider component (only for horizontal orientation right now since this is what we need atm)
- Added unit tests
- Accessibility comes from React-aria

### CODE REVIEW

File: src/components/Slider/Slider.hooks.test.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  let result;
+
+  const render = (args: SliderHookArgs) => {
+    ({ result } = renderHook((props) => useSliderSideEffects(props), {
+      initialProps: args,
+    }));
+  };
+
+  it.each([
+    {
```
Comment (jor-row) [inline src/components/Slider/Slider.hooks.test.ts:18]: question: I'm assuming it's not possible to have a value that isn't reachable with the given step? e.g. What happens if ```value: 3, minValue: 0, maxValue: 10, step: 3```?

File: src/components/Slider/Slider.types.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  /**
+   * Custom style for overriding this component's CSS.
+   */
+  style?: CSSProperties;
+
+  /**
+   * Whether Slider is disabled
+   */
+  isDisabled: boolean;
```
Comment (jor-row) [inline src/components/Slider/Slider.types.ts:24]: issue: I notice there is no styling that changes based on whether this component is disabled or not. Am I just missing it or does that need to be added? I can see that there are tokens and such for that state in the Figma

File: src/components/Slider/Slider.unit.test.tsx [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+      expect(sliderInput.getAttribute('value')).toBe(defaultProps.value.toString());
+
+      const newValue = 8;
+      rerender(<ControlledComponent newValue={newValue} />);
+
+      const sliderInputAfterChange = screen.getByRole('slider');
+      expect(sliderInputAfterChange.getAttribute('value')).toBe(newValue.toString());
+    });
+
+    it('should be change value as expected if pressing left and down arrow key', async () => {
```
Comment (jor-row) [inline src/components/Slider/Slider.unit.test.tsx:175]: nitpick: typo 
```suggestion
    it('should change value as expected when left or down arrow key is pressed', async () => {
```

File: src/components/Slider/Slider.unit.test.tsx [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+      await user.keyboard('{ArrowDown}');
+      const newValueAfterDown = newValue - defaultProps.step;
+
+      const sliderInputAfterPressingDown = screen.getByRole('slider');
+      expect(sliderInputAfterPressingDown).toHaveFocus();
+      expect(sliderInputAfterPressingDown.getAttribute('value')).toBe(newValueAfterDown.toString());
+      expect(defaultProps.onChange).toHaveBeenLastCalledWith(newValueAfterDown);
+    });
+
+    it('should be change value as expected if pressing right and up arrow key', async () => {
```
Comment (jor-row) [inline src/components/Slider/Slider.unit.test.tsx:199]: nitpick: typo
```suggestion
    it('should change value as expected when right or up arrow key is pressed', async () => {
```

File: src/components/Slider/Slider.unit.test.tsx [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+      await user.keyboard('{ArrowUp}');
+      const newValueAfterUp = newValue + defaultProps.step;
+
+      const sliderInputAfterPressingUp = screen.getByRole('slider');
+      expect(sliderInputAfterPressingUp).toHaveFocus();
+      expect(sliderInputAfterPressingUp.getAttribute('value')).toBe(newValueAfterUp.toString());
+      expect(defaultProps.onChange).toHaveBeenLastCalledWith(newValueAfterUp);
+    });
+
+    it('should be stick to minValue even after pressing more often left arrow', async () => {
```
Comment (jor-row) [inline src/components/Slider/Slider.unit.test.tsx:223]: nitpick: typo
```suggestion
    it('should stick to minValue even after pressing left arrow key more', async () => {
```

File: src/components/Slider/Slider.unit.test.tsx [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+      await user.keyboard('{ArrowLeft}');
+      const newValueAfterLeft = defaultProps.minValue;
+
+      const sliderInputAfterPressingLeft = screen.getByRole('slider');
+      expect(sliderInputAfterPressingLeft).toHaveFocus();
+      expect(sliderInputAfterPressingLeft.getAttribute('value')).toBe(newValueAfterLeft.toString());
+      expect(defaultProps.onChange).toHaveBeenLastCalledWith(newValueAfterLeft);
+    });
+
+    it('should be stick to maxValue even after pressing more often right arrow', async () => {
```
Comment (jor-row) [inline src/components/Slider/Slider.unit.test.tsx:250]: nitpick:  typo
```suggestion
    it('should stick to maxValue even after pressing right arrow key more', async () => {
```

**Review by jor-row — CHANGES_REQUESTED** [review]
A few nitpicks, questions for my benefit and one step of the build seems to be failing

File: src/components/Slider/Slider.stories.args.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      type: {
+        summary: 'function',
+      },
+      defaultValue: {
+        summary: 'undefined',
+      },
+    },
+  },
+};
+
```
Comment (jor-row) [inline src/components/Slider/Slider.stories.args.ts:77]: suggestion: Maybe adding ```isDisabled``` as an arg for the storybook would be nice

File: src/components/Slider/Slider.hooks.test.ts (hunk 1 above)
Comment (cscheffauer) [reply src/components/Slider/Slider.hooks.test.ts:18]: it is possible at the moment to set any value from the top, there is no clamping / fallback handling in place at the moment. its not necessary to have this right now for the feature we need the slider for, so i'll keep this as a tech debt, if thats fine for u!

**Review by jor-row — APPROVED** [review]

**Pull request merged**